test/e2e/prometheus_test.go

```diff
@@ -0,0 +1,279 @@
+	"antrea_controller_length_network_policy_queue",
+	"antrea_controller_network_policy_processed",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds_count",
+	"antrea_controller_network_policy_sync_duration_milliseconds_sum",
+	"antrea_controller_runtime_info",
+}
+
+// Prometheus server JSON output
+type prometheusServerOutput struct {
+	Status string
+	Data   []map[string]string
+}
+
+// Retrieve monitoring authorization token, required for access to Antrea apiserver /metrics resource
+func getMonitoringAuthToken(t *testing.T, data *TestData) string {
+	secrets, err := data.clientset.CoreV1().Secrets(monitoringNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching monitoring secret: %v", err)
+	}
+
+	if len(secrets.Items) != 1 {
+		t.Fatalf("Found unexpected number of monitoring secrets: %d", len(secrets.Items))
+	}
+	var token string
+	for _, secret := range secrets.Items {
+		token = string(secret.Data[v1.ServiceAccountTokenKey])
+	}
+
+	return token
+}
+
+// Retrieve Antrea metrics from pod apiserver
+func getMetricsFromApiServer(t *testing.T, url string, token string) string {
+	config := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+	tr := &http.Transport{TLSClientConfig: config}
+	client := &http.Client{Transport: tr}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Fatalf("Error creating HTTP request: %v", err)
+	}
+
+	if token != "" {
+		req.Header.Add("Authorization", "Bearer "+token)
+	}
+
+	// Query metrics via HTTPS from pod
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Fatalf("Error fetching metrics from %s: %v", url, err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Error retrieving metrics from %s. response: %v", url, err)
+	}
+
+	return string(body)
+}
+
+// Locate Antrea pods from the specified component, retrieve and validate that supplied metric exists
+func testPrometheusMetricsOnPods(t *testing.T, data *TestData, component string, metric string) {
+	token := getMonitoringAuthToken(t, data)
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=antrea,component=" + component,
+	}
+	pods, err := data.clientset.CoreV1().Pods(antreaNamespace).List(listOptions)
+	if err != nil {
+		t.Fatalf("Error fetching agent pods: %v", err)
+	}
+
+	var hostIP = ""
+	var hostPort int32 = 0
+
+	// Find pods' connectivity, check for metrics existence on each of them
+	for _, pod := range pods.Items {
+		hostIP = pod.Status.HostIP
+
+		for _, container := range pod.Spec.Containers {
+			for _, port := range container.Ports {
+				hostPort = port.HostPort
+				t.Logf("Found %s %d", hostIP, hostPort)
+				reqBody := getMetricsFromApiServer(t, fmt.Sprintf("https://%s:%d/metrics", hostIP, hostPort), token)
```

Comment:
seems like `respBody` would be a more accurate name here?

Reply:
Done